use-cases/azure.md

```diff
@@ -0,0 +1,319 @@
+```
+git clone https://github.com/rabbitmq/rabbitmq-server.git rabbitmq
+cd rabbitmq
+git checkout oidc-integration
+make
+```
+> :blue_book: More details about how to build RabbitMQ Server are available [here](https://www.rabbitmq.com/build-server.html)
+
+### Build RabbitMQ packages
+Building the RabbitMQ server is of limited use, unless an easy way to deploy it can be produced. 
+
+To build packages ready to be deployed, we need to clone the [rabbitmq-packaging](https://github.com/rabbitmq/rabbitmq-packaging) project at the root of the **rabbitmq-server** directory. We also need to set `RABBITMQ_PACKAGING_REPO` accordingly.
+```
+cd PATH_TO_RABBITMQ_SERVER_DIRECTORY
+git clone https://github.com/rabbitmq/rabbitmq-packaging.git
+RABBITMQ_PACKAGING_REPO=./rabbitmq-packaging make package-rpm
+```
+
+### Building RabbitMQ Docker Image
+To build the RabbitMQ docker image, you first need to build the *generic-unix* archive:
+```
+cd PATH_TO_RABBITMQ_SERVER_DIRECTORY/packaging/
+make
+```
+Then build the RabbitMQ docker image:
+```
+cd PATH_TO_RABBITMQ_SERVER_DIRECTORY/packaging/docker-images
+GENERIC_UNIX_ARCHIVE=/PATH_TO_RABBITMQ_SERVER_DIRECTORY/PACKAGES/rabbitmq-server-generic-unix-3.10.5+26.g6a3825f.tar.xz make
+```
+
+Validate that it has been created successfully:
+```
+docker images
+REPOSITORY                 TAG                      IMAGE ID       CREATED        SIZE
+pivotalrabbitmq/rabbitmq   3.10.5-26.g6a3825f       60051bf0c200   30 sec ago   307MB
+```
+### Start RabbitMQ docker container
+
+You can now **start the RabbitMQ docker container** based on the image you just built.
+
+When starting it, you need to:
+- Mount the `rabbitmq.config` file to the right location
+- Give the `rabbitmq.config` file proper permissions (the UID of rabbitmq user in the image we are going to use is **999**)
+- Set the `RABBITMQ_ADVANCED_CONFIG_FILE` environment variable to tell RabbitMQ where to find the `rabbitmq.config` file
+- Expose the RabbitMQ Management UI port (**15672**)
+
+```
+chown 999:999 ./rabbitmq.config
+docker run -p 15672:15672 \
+           -e RABBITMQ_ADVANCED_CONFIG_FILE=/etc/rabbitmq/conf.d/rabbitmq.config \
+           -v /root/rabbitmq.config:/etc/rabbitmq/conf.d/rabbitmq.config \
+           -d pivotalrabbitmq/rabbitmq:3.10.5-26.g6a3825f
+```
+
+## Start Reverse Proxy
+
+### Generate SSL certificate and key
+
+> :warning: Remember when you have registered your app on Azure AD that it only allows **https** protocol for OAuth2 **Redirect URI**? That's why we are going to deploy RabbitMQ Management UI behind a nginx Reverse Proxy, which is going to act as TLS Termination Endpoint (feel free to use another tool, or something already set-up at your place, if you prefer)
+
+For the purpose of this tutorial, you can generate a self-signed certificate/key pair (answer the questions and enter the Common Name when prompted):
+```
+openssl req -newkey rsa:2048 -nodes -keyout KEY_FILENAME.key -x509 -days 365 -out CERT_FILENAME.crt
```

Comment:
May I suggest that you create an azure folder under `bin` folder , similar to `bin/keycloak` or `bin/uaa`, called `setup` that generates this cert with the exact name you will use in the `proxy.conf`? 

In the Makefile, there could be a target "setup-azure" which calls this script. Just a suggestion.